### PR TITLE
Support CentOS Stream 9

### PR DESCRIPTION
**Pull request by elfosardo**
Tested on CentOS Stream 9 and 8 to grant retro-compatibility

### CODE REVIEW

**Comment by elfosardo** [comment]
we need python-passlib in EPEL9 for this to work https://bugzilla.redhat.com/show_bug.cgi?id=2038701

**Comment by kashifest** [comment]
/hold
Until default branch is main

**Comment by kashifest** [comment]
/hold cancel
Default branch is now main. Remember to rebase your PR on top of main branch.

**Comment by elfosardo** [comment]
this also depends on https://github.com/metal3-io/metal3-dev-env/pull/943

**Comment by kashifest** [comment]
We also need the openstack CI images and node images for target cluster to be built with CentOS stream 9 for the CI. I will take care of it.

**Comment by kashifest** [comment]
@elfosardo We are facing issues while building CI images with Centos stream 9 images. The following error is thrown by packer :

```
openstack,error []string{"Packer experienced an authentication error when trying to connect via SSH. This can happen if your username/password are wrong. You may want to double-check your credentials as part of your debugging process. original error: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain"}
==> Builds finished but no artifacts were created.
2022/03/11 11:38:21 [INFO] (telemetry) Finalizing.
Build 'openstack' errored after 1 minute 51 seconds: Packer experienced an authentication error when trying to connect via SSH. This can happen if your username/password are wrong. You may want to double-check your credentials as part of your debugging process. original error: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
```

We dont see such issues with CentOS stream 8 images. Not sure if it is related to: https://github.com/hashicorp/terraform/issues/30134 , I am planning to create an issue in packer as well.

**Comment by kashifest** [comment]
Created the issue here https://github.com/hashicorp/packer/issues/11656

**Comment by elfosardo** [comment]
next issue:
```
* minikube v1.25.2 on Centos 9
* Using the kvm2 driver based on user configuration

X Exiting due to PROVIDER_KVM2_ERROR: /usr/bin/virsh domcapabilities --virttype kvm failed:
error: failed to get emulator capabilities
error: internal error: missing 'nvram-template' in '/usr/share/qemu/firmware/50-edk2-ovmf-amdsev.json'
exit status 1
```

File: lib/common.sh [hunk 1]
```diff
@@ -53,10 +53,10 @@ source /etc/os-release
 export DISTRO="${ID}${VERSION_ID%.*}"
 export OS="${ID}"
 export OS_VERSION_ID=$VERSION_ID
-export SUPPORTED_DISTROS=(centos8 rhel8 ubuntu18 ubuntu20)
+export SUPPORTED_DISTROS=(centos8 centos9 rhel8 ubuntu18 ubuntu20)
```
Comment (dtantsur) [inline lib/common.sh:56]: Add rhel9 in advance?

**Review by dtantsur — APPROVED** [review]

File: lib/common.sh (hunk 1 above)
Comment (elfosardo) [reply lib/common.sh:56]: yeah, forgot to add that :(

**Comment by elfosardo** [comment]
/test-v1b1-centos-integration
/test-features-centos
FYI this is working locally for me with both CS8 and CS9

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Pull request merged**